### PR TITLE
Feat(enhacement): Check if a ship has the means to run from its target.

### PR DESCRIPTION
**Pull request by RisingLeaf**
## Feature Details
As requested by @Hurleveur this PR adds a small check if a ship is fast enough to keep its distance from a target.

## Testing Done
Tested fought a shield beetle with a war dragon, the shield beetle did not try to get far away.
Fought with a <insert korath big ship name i dont remember>, shield beetle kept distance.

## Performance Impact
N/A

### CODE REVIEW

File: source/AI.cpp [hunk 1]
```diff
@@ -2202,11 +2202,14 @@ void AI::Attack(Ship &ship, Command &command, const Ship &target)
 		return;
 	}
 
+	// Check if this ship is fast enough to keep distance from target.
+	bool isAbleToRun = target.MaxVelocity() < ship.MaxVelocity();
```
Comment (Hurleveur) [inline source/AI.cpp:2206]: Shouldnt we make sure its at least 10% faster or smth like that?

**Review by Hurleveur — COMMENTED** [review]
Damn thats even more simple than what I thought it'd be lol
We could consider the distance from the target, too, but that can be for a later PR, honestly this way would prob work best in most situations anyway
We may want to check for steering ability too or reversing eventually (or just later)

File: source/AI.cpp (hunk 1 above)
Comment (RisingLeaf) [reply source/AI.cpp:2206]:
```suggestion
	bool isAbleToRun = target.MaxVelocity() * 1.1 < ship.MaxVelocity();
```
like this?

**Comment by RisingLeaf** [comment]
I talked with hurl and we agreed that later we will have **the** agility calculation, that will include slowing and accel and turn. But for now it is good as is.

File: source/AI.cpp (hunk 1 above)
Comment (Hurleveur) [reply source/AI.cpp]: Sounds good

**Review by Hurleveur — APPROVED** [review]
I think this makes a lot of sense and is quite simple so its good to merge as is.

For later, we could even do a CanRunFrom(ship) method in that other agility PR, that'd be nice, and accouny for distance as well, and then the kind of weapon ~~list goes on and on~~

**Comment by EjoThims** [comment]
Seems to be even smoother distance keeping in Wah Ki, watching it on double speed for half an hour or so.

**Review by quyykk — COMMENTED** [review]
Can you add some consts? :) thanks

**Comment by RisingLeaf** [comment]
> Can you add some consts? :) thanks

you mean the 1.1? sure

**Comment by RisingLeaf** [comment]
> Can you add some consts? :) thanks

done

**Review by quyykk — APPROVED** [review]

**Pull request merged**